### PR TITLE
Fix film roll sorting order with existing setting

### LINKED ISSUE #6668: Wrong Film roll sorting order
**Describe the bug**
The sorting order of film rolls does not match the sorting order of folders if 'recent to older' is set.

**To Reproduce**
1. preferences: sort film rolls by: folder
2. preferences: sort collection recent to older: true
3. Collect by folders --> 2020 on top of the list (OK)
4. Collect by film roll --> 2020 on bottom of the list (NOK)

**Expected behavior**
If 'sort by folder' is selected, the sorting order of film rolls should match the sorting order of folders.
So if 'recent to older' is set, the newest film rolls (2020...) should be on top.

**Screenshots**
sort-filmrolls

**Platform:**
Darktable Version: 3.2.1
OS: Linux Mint 20 Cinnamon
OpenCL activated
NVIDIA Driver Version: 450.80.02

### PR DESCRIPTION
**Pull request by quovadit**
simpler way to fix #6668 (alternative to #7974)

As suggested by @elstoc in #7974 this PR uses the existing descending option to sort film-rolls

### CODE REVIEW

**Comment by elstoc** [comment]
@quovadit there's no need to keep raising new PRs when people request changes. Please close #7974 if this PR supersedes it.

**Comment by quovadit** [comment]
As I can't decide, which version will win, I created two parallel pull-requests to keep the commit-history clean.
Of course I will close the other pull requests, as soon as one gets accepted.
Sorry for the confusion...

File: data/darktableconfig.xml.in [hunk 1]
```diff
@@ -557,7 +557,7 @@
     <type>bool</type>
     <default>true</default>
     <shortdescription>sort collection recent to older</shortdescription>
-    <longdescription>changes the default collections-list order for folders, times and dates to run from recent to older</longdescription>
+    <longdescription>changes the default collections-list order for film rolls, folders, times and dates to run from recent to older</longdescription>
```
Comment (elstoc) [inline data/darktableconfig.xml.in:560]: we should change the name and description of this, possibly to just something like "sort descending" or "sort ascending" depending on which way we want the logic to work. "Recent to older" doesn't really make sense when sorting on text fields.

**Comment by elstoc** [comment]
Functionally it looks fine but I think the preference name and tooltips need to change. I prefer just one ascending/descending option rather than two, though I would prefer not to have to go into the preferences sub-menu for something that I might change quite regularly. That's something to discuss separately though.

**Comment by quovadit** [comment]
thanks again for the input, I changed name and description as suggested

**Comment by phweyland** [comment]
> Functionally it looks fine but I think the preference name and tooltips need to change. I prefer just one ascending/descending option (this PR) rather than two (#7974)

I'm aligned with that.

**Comment by phweyland** [comment]
> thanks again for the input, I changed name and description as suggested

I don't see the changes ... still "order film rolls and folders descending, times and dates from recent to older" for me ...

**Comment by quovadit** [comment]
that's the new wording, the old one was '... film rolls, folders, times and dates to run from recent to older'

**Comment by phweyland** [comment]
> that's the new wording, the old one was '... film rolls, folders, times and dates to run from recent to older'

I've missed that sorry. But I think that's the words `recent` and `older` which are not appropriate for film_roll/folder... (or you assume their name starts with the date).

**Comment by quovadit** [comment]
that's correct, that's why I tried to separate it:
- film rolls and folders descending, 
- times and dates from recent to older

Apparently that's not as clear as I thought...

**Comment by elstoc** [comment]
Would be better as "sort film rolls and folders in descending order, either by folder name or date/time stamp"

**Comment by quovadit** [comment]
in fact, this setting affects 3 types of collections:
1. 'film roll'
2. 'folders'
3. times and dates (e.g. 'date taken', export timestamp, ...)

2+3 is the same as ever, the only new field affected by this checkbox now is film-roll.

So another suggestion:
- shortdescription: sort collection descending
- longdescription: descending order for the following collections: film roll, folders, times (e.g. date-taken)

**Comment by elstoc** [comment]
> longdescription: descending order for the following collections: film roll, folders, times (e.g. date-taken)

I'd prefer if the long description was a complete sentence and the plurals were consistent, something like "sort the following types of collection in descending order: film rolls, folders, dates/times"

**Comment by quovadit** [comment]
complete sentence is fine for me

concerning plurals: I took the exact wording from the drop-down:
collections

so the longdescription would be:
sort the following collections in descending order: 'film roll', 'folders', 'times' (e.g. 'date taken')

**Comment by elstoc** [comment]
> I took the exact wording from the drop-down

Well, I would say the drop-down is inconsistent too. Based on that, "folders" should be renamed to "folder" in the drop-down and the long description should be consistent with this.

**Comment by TurboGit** [comment]
Agreed for `forlder`.

**Comment by quovadit** [comment]
changed 'folders' to 'folder' and adapted longdescription accordingly

**Comment by TurboGit** [comment]
@quovadit : please remove the po/* (translation) changes. The translator will do that especially since a plural may be translated a different way. In no case a dev should change the translation. TIA.

**Comment by quovadit** [comment]
sorry for my mistake, I reverted the translations...

**Comment by TurboGit** [comment]
I'm not fully happy. On my side I like have the film sorted with the more recent first (based on film-id) but the folder alphabetically ascending. I cannot do that now, if I select descending to get the latest film imported first I have the folder in reverse alphabetical order.

I would propose to keep the sorting by film-id always descending and sort the film using the preference only when the film is sorted based on the name.

How does that sound ?

**Comment by elstoc** [comment]
> I would propose to keep the sorting by film-id always descending

Based on the use cases described in the bug (keeping recently imported film rolls at the top) this seems reasonable to me.

**Comment by quovadit** [comment]
@TurboGit : makes sense for me, id ascending does not make any sense, and with your solution we have full flexibility with only one checkbox

File: data/darktableconfig.xml.in [hunk 2]
```diff
@@ -556,8 +556,8 @@
     <name>plugins/collect/descending</name>
     <type>bool</type>
     <default>true</default>
```
Comment (TurboGit) [inline data/darktableconfig.xml.in:558]: Sorry I see that only now, but maybe the default order should be ascending no ? And so default here set to false ?

**Comment by quovadit** [comment]
now the default is descending=true, so folders and dates are descending.
I don't see a reason to change that default.
But as the checkbox does not affect filmroll-by-id I adapted the longdescription again

**Comment by TurboGit** [comment]
Ok, merged thanks. I still find the descending order by default not really natural, but we can change that in another PR if needed.

**Pull request merged**